app/javascript/packages/document-capture/components/in-person-location-step.tsx

```diff
@@ -1,14 +1,112 @@
-import { PageHeading } from '@18f/identity-components';
-import { FormStepsButton } from '@18f/identity-form-steps';
+import { useState, useEffect } from 'react';
 import { useI18n } from '@18f/identity-react-i18n';
+import { PageHeading, LocationCollectionItem, LocationCollection } from '@18f/identity-components';
+
+interface PostOffice {
+  address: string;
+  city: string;
+  name: string;
+  phone: string;
+  saturday_hours: string;
+  state: string;
+  sunday_hours: string;
+  weekday_hours: string;
+  zip_code_4: string;
+  zip_code_5: string;
+}
+
+interface FormattedLocation {
+  addressLine2: string;
+  id: number;
+  name: string;
+  phone: string;
+  saturdayHours: string;
+  streetAddress: string;
+  sundayHours: string;
+  weekdayHours: string;
+}
+
+const locationUrl = '/verify/in_person/usps_locations';
+
+const getResponse = async () => {
+  const response = await fetch(locationUrl).then((res) =>
+    res.json().catch((error) => {
+      throw error;
+    }),
+  );
+  return response;
+};
+
+const formatLocation = (postOffices: PostOffice[]) => {
+  const formattedLocations = [] as FormattedLocation[];
+  postOffices.forEach((po: PostOffice, index) => {
+    const location = {
+      addressLine2: `${po.city}, ${po.state}, ${po.zip_code_5}-${po.zip_code_4}`,
+      id: index,
+      name: po.name,
+      phone: po.phone,
+      saturdayHours: po.saturday_hours,
+      streetAddress: po.address,
+      sundayHours: po.sunday_hours,
+      weekdayHours: po.weekday_hours,
+    } as FormattedLocation;
+    formattedLocations.push(location);
+  });
+  return formattedLocations;
+};
 
 function InPersonLocationStep() {
   const { t } = useI18n();
+  const [locationData, setLocationData] = useState([] as FormattedLocation[]);
+
+  const handleLocationSelect = async (id: number) => {
```

Comment:
This writes the location to session via `UspsLocationsController`, and the following step, the `InPersonPrepareStep`, reads it from the session. It's currently possible for the read to happen before write is finished, so we need to wait until the write is finished to continue to the next page.

Reply:
fixed in b06bf631306c4512e692ae67ee320d85bc30288d